tests/django/test_model_mockers.py

```diff
@@ -0,0 +1,189 @@
+import json
+
+import pytest
+from django.apps import apps
+from django.contrib.gis.db.models import fields as geo_fields
+from django.db.models import fields
+from factory.base import FactoryMetaClass
+
+from schematools.contrib.django.db import create_tables
+from schematools.contrib.django.factories import (
+    model_mocker_factory,
+    schema_model_mockers_factory,
+    schema_models_factory,
+)
+from schematools.contrib.django.models import Dataset
+from schematools.types import DatasetTableSchema
+
+
+@pytest.mark.django_db
+def test_model_mocker_factory_registers_a_dynamic_model_in_the_app_config(afval_dataset) -> None:
+    """Prove that the model_mocker_factory registers the model as Django app."""
+    table_schema = afval_dataset.schema.tables[0]
+    assert isinstance(table_schema, DatasetTableSchema)
+    assert str(table_schema) == "<DatasetTableSchema: afvalwegingen.containers>"
+
+    # We bootstrap the DynamicModel inside the model_mocker_factory by calling the model_factory,
+    # which ends with app_config.register_model.
+    # So we assert that the DynamicModel is indeed registered.
+    model_mocker_factory(afval_dataset, table_schema, base_app_name="dso_api.dynamic_api")
+    registered_models = [f"{m._meta.app_label}.{m._meta.model_name}" for m in apps.get_models()]
+    assert "afvalwegingen.containers" in registered_models
+
+    # The afval_dataset has two tables (containers, clusters),
+    # so we check the second table (i.e. second model) too.
+    table_schema_with_id_as_string = afval_dataset.schema.tables[1]
+    assert isinstance(table_schema_with_id_as_string, DatasetTableSchema)
+    assert str(table_schema_with_id_as_string) == "<DatasetTableSchema: afvalwegingen.clusters>"
+
+    model_mocker_factory(
+        afval_dataset, table_schema_with_id_as_string, base_app_name="dso_api.dynamic_api"
+    )
+    registered_models_updated = [
+        f"{m._meta.app_label}.{m._meta.model_name}" for m in apps.get_models()
+    ]
+    assert "afvalwegingen.clusters" in registered_models_updated
```

Comment:
Wat test dit nu eigenlijk? Bestaande functionaliteit, en het lijt niet echt iets met de mockers te tetsen. De laatste check had ook heel goed al true kunnen zijn voordat de mocker gestart werd. Dat moet iig zeker nagekeken worden

Reply:
Ja, mee eens. Dit zat in het eerste opzetje van Timo, maar was denk ik voornamelijk copy/paste. Kan er denk ik wel (grotendeels) uit.